tests/framework/matcher/restarted.go

```diff
@@ -0,0 +1,55 @@
+package matcher
+
+import (
+	"fmt"
+
+	"github.com/onsi/gomega/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/framework/matcher/helper"
+)
+
+func BeRestarted(oldUID k8stypes.UID) types.GomegaMatcher {
+	return restartedMatcher{
+		oldUID: oldUID,
+	}
+}
+
+type restartedMatcher struct {
+	oldUID k8stypes.UID
+}
+
+func (r restartedMatcher) Match(actual interface{}) (success bool, err error) {
+	u, err := helper.ToUnstructured(actual)
```

Comment:
Do you expect other types of `actual` than `VirtualMachineInstance`? If not, maybe you can change this to directly typecast to VMI and use it, instead of using unstructered.